src/site-server.ts

```diff
@@ -86,7 +98,16 @@ export class SiteServer {
 			siteTitle: this.details.name,
 			php: this.details.phpVersion,
 		} );
-		const absoluteUrl = `http://localhost:${ this.details.port }`;
+		// Determine the URL to use - either custom domain or localhost with port
+		let absoluteUrl;
+		if ( this.details.useCustomDomain && this.details.customDomain ) {
+			absoluteUrl = `http://${ this.details.customDomain }`;
```

Comment:
Maybe it's worth calling `addDomainToHosts` here, in case the hosts definitions were modified?